### PR TITLE
Review suggestions. Update code to reflect suggestions

### PR DESCRIPTION
**Pull request by vikas-bishnoi**
Review suggestions in create contacts and list contacts UI and changed code as suggested

### CODE REVIEW

File: app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx [hunk 1]
```diff
@@ -1,41 +1,48 @@
...
+  };
+
   return (
     <Formik
       initialValues={contact}
       validateOnBlur={submitted}
       validateOnChange={submitted}
       validationSchema={CONTACTS_FORM_VALIDATION_SCHEMA}
-      onSubmit={() => {}}
+      onSubmit={() => handleSubmit()}
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx:28]: Use the shorthand notation whenever possible.

```suggestion
onSubmit={handleSubmit}
```

File: app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx [hunk 2]
```diff
@@ -73,7 +80,8 @@ const ContactForm = ({ onClose, contact }) => {
               label="Cancel"
               size="large"
               style="text"
-              onClick={onClose}
+              type="reset"
+              onClick={() => onClose()}
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx:84]: Refer prev comment.

File: app/javascript/src/components/Dashboard/Contacts/constants.jsx [hunk 3]
```diff
@@ -9,14 +9,21 @@ export const CONTACTS_FORM_INITIAL_FORM_VALUES = {
...
   first_name: yup.string().required("First Name is required"),
   last_name: yup.string().required("Last Name is required"),
   email: yup.string().email("Invalid email").required("Email is required"),
-  role: yup.string().required("Role is required"),
+  role: yup
+    .object()
+    .nullable()
+    .shape({
+      label: yup.string(),
+      value: yup.string(),
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/constants.jsx:24]: Refer the comments in Slack thread. There was a suggestion regarding using `oneOf`.

File: app/javascript/src/components/Dashboard/Notes/constants.js [hunk 4]
```diff
@@ -4,13 +4,29 @@ import * as yup from "yup";
...
+    })
+    .required("Assigned Contact is required"),
+  tag: yup
+    .object()
+    .nullable()
+    .shape({
+      label: yup.string(),
+      value: yup.string(),
+    })
+    .required("Tag is required"),
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/constants.js:29]: Why exactly is that in your business logic you are not allowing a `note` to have multiple `tags`?

Example:
- https://handbook.neetokb.com/articles/git-and-github-essentials
- ^ This note has both the `git` and `github` tags.

**Review by yedhink — COMMENTED** [review]
@vikas-bishnoi

File: app/javascript/src/components/Dashboard/Notes/constants.js (hunk 4 above)
Comment (vikas-bishnoi) [reply app/javascript/src/components/Dashboard/Notes/constants.js:29]: This is resolved.
I have set added isMulti prop to select. Also, I have updated validation for multiple tags using an array of objects in yup validation schema.

File: app/javascript/src/components/Dashboard/Contacts/constants.jsx (hunk 3 above)
Comment (vikas-bishnoi) [reply app/javascript/src/components/Dashboard/Contacts/constants.jsx:24]: I have resolved this using and used oneOf to validate label and value

File: app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx (hunk 2 above)
Comment (vikas-bishnoi) [reply app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx:84]: I have resolved this.

File: app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx (hunk 1 above)
Comment (vikas-bishnoi) [reply app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx:28]: I have resolved this.

**Pull request merged**